src/main/java/seedu/mark/logic/commands/AddCommand.java

```diff
@@ -35,21 +36,42 @@
 
     public static final String MESSAGE_SUCCESS = "New bookmark added: %1$s";
     public static final String MESSAGE_DUPLICATE_BOOKMARK = "This bookmark already exists in Mark";
+    public static final String MESSAGE_MISSING_URL = "Failed to obtain the URL of the current page.";
 
-    private final Bookmark toAdd;
+    private Bookmark toAdd;
+    private boolean hasNoUrl;
 
     /**
-     * Creates an AddCommand to add the specified {@code Bookmark}
+     * Creates an AddCommand to add the specified {@code Bookmark}, the Url
+     * of the bookmark must be present.
      */
     public AddCommand(Bookmark bookmark) {
+        this(bookmark, false);
+    }
+
+    /**
+     * Creates an AddCommand to add the specified {@code Bookmark}
+     */
+    public AddCommand(Bookmark bookmark, boolean hasNoUrl) {
         requireNonNull(bookmark);
-        toAdd = bookmark;
+        this.toAdd = bookmark;
+        this.hasNoUrl = hasNoUrl;
+
     }
 
     @Override
     public CommandResult execute(Model model, Storage storage) throws CommandException {
         requireAllNonNull(model, storage);
 
+        if (hasNoUrl) {
+            Url urlToAdd = model.getCurrentUrl();
+            if (urlToAdd == null) {
+                throw new CommandException(MESSAGE_MISSING_URL);
+            }
+            toAdd = new Bookmark(toAdd.getName(), urlToAdd, toAdd.getRemark(), toAdd.getFolder(),
```

Comment:
```suggestion
            toAdd = new BookmarkBuilder(toAdd).with(urlToAdd).build();
```
May be easier to read. And also need to import BookmarkBuilder.